### PR TITLE
[feature] Merge prereg-prize-2 into prereg-prize branch

### PR DESCRIPTION
**Pull request by HarryRybacki-zz**
## Purpose

Merge `feature/prereg-prize-2` into `feature/prereg-prize`
This merge will bring in the actual prereg-prize specific features that have been developed by @samchrisinger and @barbour-em thus far.
## Changes
- **Activate** previously commented out code related to prereg-specific functionality
- Adds `prereg-prize-test.json` schema to existing set of schemas
- Expands `MetaSchema` to accommodate prereg-prize needs
- Adds `DraftRegistrationApproval` class
- Adds `draft approval` route
- Adds numerous docstrings related to prereg and registration code
## Notes:

This will require some thorough regression testing to make sure we don't re-work any existing functionality we don't have 'checked off' yet.
1. There seems to be a PUT request for each character modified on field inputs. Is this excessive/necessary?
   - This is causing console errors when attempting to reach `/api/v1/project/<pid>/draft/<draft_id>/ endpoints. Server logs show django related errors.
## Prereg-prize draft registration overview:
- Template
  screen shot 2015-09-21 at 10 49 48 am
  - **Fulfulls** section links to [https://cos.io/prereg](prereg-prize marketing page).  
  - **TODO**: Fix html being escaped in the prereg-prize template
- Tooltip regarding **Requires Approval** warning
  screen shot 2015-09-21 at 10 50 02 am
- In template examples
  screen shot 2015-09-21 at 12 13 46 pm
  screen shot 2015-09-21 at 12 13 53 pm
- In template comments for each form field
  screen shot 2015-09-21 at 12 14 41 pm
- **Submit for review** button
  - Overview
    screen shot 2015-09-21 at 12 18 23 pm
  - Upon clicking **Submit for review**
    screen shot 2015-09-21 at 12 26 01 pm
  - After confirming submission
    screen shot 2015-09-21 at 12 26 09 pm
- File uploads for specific form fields
  - Not expanded
    screen shot 2015-09-21 at 12 19 07 pm
  - Expanded
    screen shot 2015-09-21 at 12 21 32 pm
  - **NOTE:** files are attached via files that have been added to the parent project -- not something a user would upload from their computer.
- **Certify** user input seems wonky.
  - Do we want the user to 'sign' the text input? If so we should expand the placeholder to be more explicit.
    screen shot 2015-09-21 at 12 23 38 pm

### CODE REVIEW

**Comment by lyndsysimon** [comment]
Could you verify for me that file uploads, if they are in this PR, are working as intended?

I ask because I'm wanting to make sure we don't have any unforeseen merge issues with the new file model hierarchy. I didn't see anything in the changes that was concerning on a quick glance, for what that's worth.

**Comment by HarryRybacki-zz** [comment]
@lyndsysimon file uploads to work. They aren't uploads in the classic sense that a user will _upload a file from their machine._ Rather the user can associate a file uploaded to the _project_ a registration is _based off of._ It uses the same file view that `/files/` will provide so it should feel familiar to regular users.

File: website/static/js/registrationUtils.js [hunk 1]
```diff
@@ -32,59 +32,60 @@ var currentUser = window.contextVars.currentUser || {
...
-      else {
-        return self.user.fullname;
-      }
-    });
+        self.seenBy = ko.observableArray([self.user.id] || []);
+        /**
+         * Returns the author as the actual user, not 'You'
+         **/
+        self.author = ko.pureComputed(function() {
+            return self.user.fullname;
```
Comment (HarryRybacki-zz) [inline website/static/js/registrationUtils.js:61]: if `self.user` comes from `currentUser` but contextVars isn't populated with a user, there won't be a fullname field

Is that even possible?

Comment (atelic) [inline website/static/js/registrationUtils.js:61]: @HarryRybacki this shouldn't be possible. A user will have to be a contributor on the project and logged in to comment or to access the form.

**Comment by atelic** [comment]
## Some notes
1. The rationale behind the PUT for every `keyup` was the emulation of an autosave feature based off discussion with community. I haven't seen the console errors so I will look into that soon. 
2. @HarryRybacki can you confirm that pressing enter anywhere on the form page causes a new comment to be created? If this is the case then [this PR](https://github.com/samchrisinger/osf.io/pull/38) will need to be added 
3. Judging from the language on JIRA, it seems we have started calling this the Prereg Challenge, if this is a global change the after submission modal and schema should reflect this.
4. Re file uploads, there was talk of making it so that if the user wanted to upload a file that was not part of the parent project, the upload widget would upload it to the project and then 'link' it like it does. Is this the current behavior?
5. I agree that the **certify** fields do seem off and break user flow. Not sure what the community requirements for this are though

**Comment by HarryRybacki-zz** [comment]
Moved to [PR against Center for Open Science prereg feature branch](https://github.com/CenterForOpenScience/osf.io/pull/4338).

**Pull request closed without merging**